### PR TITLE
More changes to splices data model

### PR DESCRIPTION
**Pull request by t-bast**
This is a collection of small independent commits on top of #2540

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala [hunk 1]
```diff
@@ -826,7 +826,7 @@ case class Commitments(channelId: ByteVector32,
   /**
    * When reconnecting, we drop all unsigned changes.
    */
-  def discardUnsignedUpdates(implicit log: LoggingAdapter): Commitments = {
+  def discardUnsignedUpdates()(implicit log: LoggingAdapter): Commitments = {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala:829]: Actually this method can be removed entirely, we only need it at the `MetaCommitments` level

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/MetaCommitments.scala [hunk 2]
```diff
@@ -38,19 +38,21 @@ case class Common(localChanges: LocalChanges, remoteChanges: RemoteChanges,
                   originChannels: Map[Long, Origin], // for outgoing htlcs relayed through us, details about the corresponding incoming htlcs
                   remoteNextCommitInfo: Either[WaitForRev, PublicKey], // this one is tricky, it must be kept in sync with Commitment.nextRemoteCommit_opt
                   remotePerCommitmentSecrets: ShaChain) {
+  val nextRemoteCommitIndex = remoteCommitIndex + 1
+
   /**
    * When reconnecting, we drop all unsigned changes.
    */
-  def discardUnsignedUpdates(implicit log: LoggingAdapter): Common = {
+  def discardUnsignedUpdates()(implicit log: LoggingAdapter): Common = {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/channel/MetaCommitments.scala:46]: I don't think we should add parentheses, as there is no side effect?

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala [hunk 3]
```diff
@@ -381,8 +379,8 @@ trait ChannelOpenDualFunded extends DualFundingHandlers with ErrorHandlers {
 
   when(WAIT_FOR_DUAL_FUNDING_CONFIRMED)(handleExceptions {
     case Event(txSigs: TxSignatures, d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) =>
-      d.metaCommitments.main.localFundingStatus match {
-        case DualFundedUnconfirmedFundingTx(fundingTx: PartiallySignedSharedTransaction) => InteractiveTxBuilder.addRemoteSigs(d.fundingParams, fundingTx, txSigs) match {
+      d.metaCommitments.main.localFundingStatus.asInstanceOf[DualFundedUnconfirmedFundingTx].sharedTx match {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala:382]:
```suggestion
      d.mainFundingTx match {
```

File: eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala [hunk 4]
```diff
@@ -309,17 +309,20 @@ private[channel] object ChannelCodecs3 {
...
 
+    private val dualFundedUnconfirmedFundingTxCodec: Codec[DualFundedUnconfirmedFundingTx] = (
+      ("sharedTx" | signedSharedTransactionCodec) ::
+        ("createdAt" | blockHeight)).as[DualFundedUnconfirmedFundingTx]
+
     val fundingTxStatusCodec: Codec[LocalFundingStatus] = discriminated[LocalFundingStatus].by(uint8)
       .typecase(0x00, optional(bool8, txCodec).as[SingleFundedUnconfirmedFundingTx])
       .typecase(0x01, txCodec.map(tx => SingleFundedUnconfirmedFundingTx(Some(tx))).decodeOnly) // for backward compat
-      .typecase(0x02, signedSharedTransactionCodec.as[DualFundedUnconfirmedFundingTx])
+      .typecase(0x02, dualFundedUnconfirmedFundingTxCodec)
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala:319]: Did you forget to handle backward compatibility? Would a blockheight of zero make sense?

**Review by pm47 — COMMENTED** [review]
Changes have been cherry-picked ton #2540, with the following amendment: fcf5698f0.

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/MetaCommitments.scala (hunk 2 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/channel/MetaCommitments.scala:46]: It's very weird without parenthesis, `discardXXX` really feels like it's _doing_ something, not just _accessing_ something...

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala (hunk 3 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala:382]: Nice!

File: eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala (hunk 4 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala:319]: I didn't realize you had been clever to make it work with backwards-compat! Yes a blockheight of zero totally makes sense, it doesn't really matter if current data isn't migrated to meaningful values.

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/MetaCommitments.scala (hunk 2 above)
Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/channel/MetaCommitments.scala:46]: I put them back in 612b327a8.

**Pull request closed without merging**